rest/taskana-rest-spring/src/main/java/pro/taskana/rest/AccessIdController.java

```diff
@@ -35,28 +35,37 @@
 
     @GetMapping
     public ResponseEntity<List<AccessIdResource>> validateAccessIds(
-        @RequestParam String searchFor, @RequestParam(required = false) boolean searchInGroups) throws InvalidArgumentException {
+        @RequestParam("search-for") String searchFor) throws InvalidArgumentException {
         if (searchFor.length() < ldapClient.getMinSearchForLength()) {
-            throw new InvalidArgumentException("searchFor string '" + searchFor + "' is too short. Minimum searchFor length = "
-                + ldapClient.getMinSearchForLength());
+            throw new InvalidArgumentException(
+                "searchFor string '" + searchFor + "' is too short. Minimum searchFor length = "
+                    + ldapClient.getMinSearchForLength());
         }
         if (ldapClient.useLdap()) {
             List<AccessIdResource> accessIdUsers = ldapClient.searchUsersAndGroups(searchFor);
-            if (searchInGroups) {
-                List<AccessIdResource> accessIdGroups = ldapClient.searchGroupsofUsersIsMember(searchFor);
-                accessIdUsers.addAll(accessIdGroups);
-            }
             return new ResponseEntity<>(accessIdUsers, HttpStatus.OK);
         } else if (ldapCache != null) {
-            if (searchInGroups) {
-                return new ResponseEntity<>(
-                        ldapCache.findGroupsOfUser(searchFor, ldapClient.getMaxNumberOfReturnedAccessIds()),
-                        HttpStatus.OK);
-            } else {
-                return new ResponseEntity<>(
-                        ldapCache.findMatchingAccessId(searchFor, ldapClient.getMaxNumberOfReturnedAccessIds()),
-                        HttpStatus.OK);
-            }
+            return new ResponseEntity<>(
+                ldapCache.findMatchingAccessId(searchFor, ldapClient.getMaxNumberOfReturnedAccessIds()),
+                HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(new ArrayList<>(), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @GetMapping(path = "/get-groups")
```

Comment:
We should always use an entity for the path. In this case 'groups'. GET is the HTTP method.

Reply:
Done